.github/workflows/terraform_checks.yml

```diff
@@ -0,0 +1,41 @@
+---
+name: Infra Checks
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - "ops/**"
+env:
+  ARM_CLIENT_ID: ${{ secrets.TERRAFORM_ARM_CLIENT_ID }}
+  ARM_CLIENT_SECRET: ${{ secrets.TERRAFORM_ARM_CLIENT_SECRET }}
+  ARM_SUBSCRIPTION_ID: ${{ secrets.TERRAFORM_ARM_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.TERRAFORM_ARM_TENANT_ID }}
+
+jobs:
+  validation_and_formatting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1
+
+      - name: Terraform fmt
+        id: fmt
+        working-directory: ${{ github.workspace }}/ops
+        run: |
+          terraform fmt -check dev
+          terraform fmt -check global
+
+      - name: Terraform Init
+        id: init
+        working-directory: ${{ github.workspace }}/ops
+        run: |
+          cd dev && terraform init
+          cd ../global && terraform init
+
+      - name: Terraform Validate
+        id: validate
+        working-directory: ${{ github.workspace }}/ops
```

Comment:
Seems like this should just be done the same way as it is in the deploy jobs, no?

    defaults:
      run:
        working-directory: ./ops